src/plugins/ovp/services/JwplayerMediaSubscriber.js

```diff
@@ -0,0 +1,71 @@
+import EventSubscriber from '@gdbots/pbjx/EventSubscriber';
+import getTextFieldError from '@triniti/cms/components/text-field/getTextFieldError';
+import Microtime from '@gdbots/pbj/well-known/Microtime';
+
+export default class JwplayerMediaSubscriber extends EventSubscriber {
+  constructor() {
+    super();
+    this.onInitForm = this.onInitForm.bind(this);
+    this.onValidateForm = this.onValidateForm.bind(this);
+    this.onSubmitForm = this.onSubmitForm.bind(this);
+  }
+
+  /**
+   * Runs when the form is first created and is used to
+   * update initial values.
+   *
+   * @param {FormEvent} formEvent
+   */
+  onInitForm(formEvent) {
+    const data = formEvent.getData();
+    const node = formEvent.getMessage();
+
+    data.jwplayerMediaId = node.get('jwplayer_media_id');
+    data.jwplayerSyncEnabled = node.get('jwplayer_sync_enabled');
+    data.jwplayerSyncedAt = node.has('jwplayer_synced_at')
+      ? Microtime.fromString(String(`${node.get('jwplayer_synced_at', 1574813280)}000000`)).toMoment().format('MMM DD, YYYY hh:mm A')
+      : null;
+  }
+
+  /**
+   * This is the redux form sync validation process.
+   *
+   * @param {FormEvent} formEvent
+   */
+  onValidateForm(formEvent) {
+    const data = formEvent.getData();
+    const node = formEvent.getMessage();
+
+    const error = getTextFieldError(data, 'jwplayerMediaId', node);
+    if (error) {
+      formEvent.addError('jwplayerMediaId', error);
+    }
+  }
+
+  /**
+   * Binds data from the redux form to the command.
+   * This occurs AFTER a form has been submitted
+   * but before the command is dispatched.
+   *
+   * @param {FormEvent} formEvent
+   */
+  onSubmitForm(formEvent) {
+    const data = formEvent.getData();
+    const node = formEvent.getMessage();
+
+    if (node.isFrozen()) {
+      return;
+    }
+
+    node.set('jwplayer_media_id', data.jwplayerMediaId || null);
+    node.set('jwplayer_sync_enabled', data.jwplayerSyncEnabled);
```

Comment:
@efjacobson does this cast to bool:false correctly when not enabled?

Reply:
@gdbrown yes, because it is first initialized as a bool and then can only toggle between true/false